src/cmd/singularity/cli/build.go

```diff
@@ -73,52 +72,109 @@ var BuildCmd = &cobra.Command{
 			fmt.Println("Sandbox!")
 		}
 
-		if isJSON {
-			b, err = build.NewSIFBuilderJSON(args[0], strings.NewReader(args[1]))
-			if err != nil {
-				sylog.Fatalf("Unable to parse JSON: %v\n", err)
+		def := makeDefinition(args[1], isJSON)
+
+		if remote || builderURL != defbuilderURL {
+			// Submiting a remote build requires a valid authToken
+			var b *build.RemoteBuilder
+			if authToken != "" {
+				b = build.NewRemoteBuilder(args[0], libraryURL, def, detached, builderURL, authToken)
+			} else {
+				sylog.Fatalf("Unable to submit build job: %v", authWarning)
 			}
+			b.Build(context.TODO())
+
 		} else {
-			if ok, err := build.IsValidURI(args[1]); ok && err == nil {
-				// URI passed as arg[1]
-				def, err = build.NewDefinitionFromURI(args[1])
-				if err != nil {
-					sylog.Fatalf("unable to parse URI %s: %v\n", args[1], err)
-				}
-			} else if !ok && err == nil {
-				// Non-URI passed as arg[1]
-				defFile, err := os.Open(args[1])
-				if err != nil {
-					sylog.Fatalf("unable to open file %s: %v\n", args[1], err)
-				}
-				defer defFile.Close()
-
-				def, err = build.ParseDefinitionFile(defFile)
-				if err != nil {
-					sylog.Fatalf("failed to parse definition file %s: %v\n", args[1], err)
-				}
+			//local build
+			bundle := makeBundle(def)
+
+			if sandbox {
+				sylog.Fatalf("Cannot build to sandbox... yet")
 			} else {
-				sylog.Fatalf("unable to parse %s: %v\n", args[1], err)
+				a = &build.SIFAssembler{}
 			}
 
-			if remote || builderURL != defbuilderURL {
-				// Submiting a remote build requires a valid authToken
-				if authToken != "" {
-					b = build.NewRemoteBuilder(args[0], libraryURL, def, detached, builderURL, authToken)
-				} else {
-					sylog.Fatalf("Unable to submit build job: %v", authWarning)
-				}
-			} else {
-				b, err = build.NewSIFBuilder(args[0], def)
-				if err != nil {
-					sylog.Fatalf("failed to create SifBuilder object: %v\n", err)
-				}
+			err = a.Assemble(bundle, args[0])
+			if err != nil {
+				sylog.Fatalf("Assembler failed to assemble:", err)
 			}
+			// if remote || builderURL != defbuilderURL {
+			// 	// Submiting a remote build requires a valid authToken
+			// 	if authToken != "" {
+			// 		b = build.NewRemoteBuilder(args[0], libraryURL, def, detached, builderURL, authToken)
+			// 	} else {
+			// 		sylog.Fatalf("Unable to submit build job: %v", authWarning)
+			// 	}
+			// } else {
+			// 	b, err = build.NewSIFBuilder(args[0], def)
+			// 	if err != nil {
+			// 		sylog.Fatalf("failed to create SifBuilder object: %v\n", err)
+			// 	}
+			// }
```

Comment:
Can we remove these commented-out lines? I don't think it's adding much (can always look in the `git` history.)

Reply:
Whoops... one sec